ctags/parsers/autoit.c

```diff
@@ -0,0 +1,121 @@
+/*
+*   Copyright (c) 2017, Alexey Olenich
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License version 2 or (at your option) any later version.
+*
+*   This module contains functions for generating tags for AutoIt functions.
+*   Homepage             https://www.autoitscript.com/site/autoit/
+*   Online Documentation https://www.autoitscript.com/autoit3/docs/
+*
+*   Functions: (?i)^func[ \t]{1,}([a-z0-9_]{1,}) >> \1
+*   Regions:   (?i)^#region[ \t]*(.*?)           >> \1
+*/
+
+/*
+*   INCLUDE FILES
+*/
+#include "general.h"  /* must always come first */
+
+#include <string.h>
+
+#include "parse.h"
+#include "read.h"
+#include "routines.h"
+#include "vstring.h"
+
+/*
+*   DATA DEFINITIONS
+*/
+typedef enum {
+	K_FUNCTION,
+	K_REGION
+} AutoItKind;
+
+static kindOption AutoItKinds [] = {
+	{ true, 'f', "func", "functions" },
+	{ true, 'r', "region", "regions" }
+};
+
+/*
+*   FUNCTION DEFINITIONS
+*/
+static void findAutoItTags (void)
+{
+	vString *name = vStringNew ();
+	const unsigned char *line;
+
+	while ((line = readLineFromInputFile ()) != NULL)
+	{
+		const unsigned char* p = line;
+		if (p [0] == '#')
+		{
+			/* min. string "#region" > 7 */
+			if ((p [1] == 'R' || p [1] == 'r') &&
+				strlen ((const char *) p) > 8 &&
+				(p [2] == 'E' || p [2] == 'e') &&
+				(p [3] == 'G' || p [3] == 'g') &&
+				(p [4] == 'I' || p [4] == 'i') &&
+				(p [5] == 'O' || p [5] == 'o') &&
+				(p [6] == 'N' || p [6] == 'n'))
```

Comment:
Valid syntax:
```autoit
#Region
#Region ;foobar something
#Region;foobar something
#Region foobar something
```
In other cases string will be ignored (and of course in first case it will not be shown in the list of symbols too). Works fine, I use it long time ago.

Reply:
Hum, I just tested whether I'm wasn't crazy, and I ain't:
```autoit
#regiontest
#endregion
```
results in showing a region named `test` in the symbols tree.

It might be considered OK as this is probably invalid syntax, I'm just mentioning this so you can make an informed decision whether it's a problem or an acceptable behavior given invalid input not worth handling.

---
Additionally with your examples, the leading `;` is included in the displayed region name, is that intentional? If not, you probably could simply use
```c
while (isspace ((int) *p) || *p == ';')
```
below.  It would allow for `#region;;;;foo`, but that might be valid or OK to get wrong (as suggested above, it can be considered OK to emit improper tags in invalid input)

Reply:
I'm not sure, but I actually expect the `;` to start a standard comment, so it's probably an unnamed region, with just a comment that happens to be there.

Reply:
I think yes, most likely, but it's used, it works (in other editors too), it's not a glaring mistake and I decided to do so.